src/lua/protobuf_wireformat.lua

```diff
@@ -97,6 +141,29 @@ local function encode_varint(value)
     return ffi.string(buf, size)
 end
 
+-- Decode input value as VARINT without a tag.
+--
+-- Input value type: string
+--
+-- This is a helper function to decode tag and data values.
+local function decode_varint(ctx)
+    local msb
+    local res = 0
+    local len = 0
+    repeat
+        local payload = ffi.cast('uint64_t', string.byte(ctx.val, ctx.pos))
+        if len >= 10 or payload == nil then
```

Comment:
How `payload` can be `nil`?

Reply:
This check was implemented earlier to raise an error when decoder meets the end of the line while decoding varint value. It seems like this check does not work as intended. Added a function to read known amount of bytes from decoded string with boundary check.